src/elbo.jl

```diff
@@ -1,18 +1,21 @@
 function maximize_elbo(rng, logp, dists, ndraws, executor)
+    μ = dists[1].μ
+    N = length(μ)
+    u = Random.randn!(rng, similar(μ, N, ndraws))
     EE = Core.Compiler.return_type(
-        elbo_and_samples, Tuple{typeof(rng),typeof(logp),eltype(dists),Int}
+        elbo_and_samples!, Tuple{typeof(rng),typeof(u),typeof(logp),eltype(dists),Int}
     )
     estimates = similar(dists, EE)
     isempty(estimates) && return 0, estimates
     Folds.map!(estimates, dists, executor) do dist
```

Comment:
Since `u` is being overwritten here, I think we cannot use Transducers, or different threads could be writing to it at the same time. This will probably just need to be made a simple `map`